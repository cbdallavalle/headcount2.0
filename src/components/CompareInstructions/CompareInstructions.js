import React from 'react';
import Card from '../Card/Card';
import PropTypes from 'prop-types';
import arrow from '../../images/arrow2.gif';


const CompareInstructions = ({ comparisonData, handleCompareCards }) => {
  return (
    <section className="displayOne">
      <Card 
        schoolData={ comparisonData[0] }
        handleCompareCards={ handleCompareCards }
        size='small'
        clicked='clicked'
      />
      <div className="instructions-cont">
        <p className='instructions'>
          click another to compare
        </p>
        <img src={ arrow } alt="Choose a Card To Compare With" id="arrow"/>
      </div>
      <div className='shadow-card'>
      </div>
    </section>
  );
};

const { array, func } = PropTypes;

CompareInstructions.propTypes = {
  comparisonData: array.isRequired,
  handleCompareCards: func.isRequired
};

export default CompareInstructions;