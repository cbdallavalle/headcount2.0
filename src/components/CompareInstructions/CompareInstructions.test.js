/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import CompareInstructions from './CompareInstructions';

it('should match the snapshot', () => {
  const mockedComparisonData = {
    school1: {
      data: {2004: 0.302, 2005: 0.267, 2006: 0.354, 2007: 0.392, 2008: 0.385,
        2009: 0.39, 2010: 0.436, 2011: 0.489, 2012: 0.479, 2013: 0.488, 2014: 0.49},
      dataType: "Percent",
      location: "ACADEMY 20"
    }
  }
  const wrapper = shallow(<CompareInstructions
    comparisonData={mockedComparisonData}
    handleCompareCards={() => {}}
  />)
  expect(wrapper).toMatchSnapshot();
})