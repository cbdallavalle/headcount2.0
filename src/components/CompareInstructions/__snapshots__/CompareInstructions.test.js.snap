// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CompareInstructions
    comparisonData={
        Array [
            Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
          ]
    }
    handleCompareCards={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          clicked="clicked"
          handleCompareCards={[Function]}
          schoolData={
                    Object {
                              "data": Object {
                                "2004": 0.302,
                                "2005": 0.267,
                                "2006": 0.354,
                                "2007": 0.392,
                                "2008": 0.385,
                                "2009": 0.39,
                                "2010": 0.436,
                                "2011": 0.489,
                                "2012": 0.479,
                                "2013": 0.488,
                                "2014": 0.49,
                              },
                              "dataType": "Percent",
                              "location": "ACADEMY 20",
                            }
          }
          size="small"
/>,
        <div
          className="instructions-cont"
>
          <p
                    className="instructions"
          >
                    click another to compare
          </p>
          <img
                    alt="Choose a Card To Compare With"
                    id="arrow"
                    src="arrow2.gif"
          />
</div>,
        <div
          className="shadow-card"
/>,
      ],
      "className": "displayOne",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "clicked": "clicked",
          "handleCompareCards": [Function],
          "schoolData": Object {
            "data": Object {
              "2004": 0.302,
              "2005": 0.267,
              "2006": 0.354,
              "2007": 0.392,
              "2008": 0.385,
              "2009": 0.39,
              "2010": 0.436,
              "2011": 0.489,
              "2012": 0.479,
              "2013": 0.488,
              "2014": 0.49,
            },
            "dataType": "Percent",
            "location": "ACADEMY 20",
          },
          "size": "small",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="instructions"
>
              click another to compare
</p>,
            <img
              alt="Choose a Card To Compare With"
              id="arrow"
              src="arrow2.gif"
/>,
          ],
          "className": "instructions-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "click another to compare",
              "className": "instructions",
            },
            "ref": null,
            "rendered": "click another to compare",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Choose a Card To Compare With",
              "id": "arrow",
              "src": "arrow2.gif",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "shadow-card",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            clicked="clicked"
            handleCompareCards={[Function]}
            schoolData={
                        Object {
                                    "data": Object {
                                      "2004": 0.302,
                                      "2005": 0.267,
                                      "2006": 0.354,
                                      "2007": 0.392,
                                      "2008": 0.385,
                                      "2009": 0.39,
                                      "2010": 0.436,
                                      "2011": 0.489,
                                      "2012": 0.479,
                                      "2013": 0.488,
                                      "2014": 0.49,
                                    },
                                    "dataType": "Percent",
                                    "location": "ACADEMY 20",
                                  }
            }
            size="small"
/>,
          <div
            className="instructions-cont"
>
            <p
                        className="instructions"
            >
                        click another to compare
            </p>
            <img
                        alt="Choose a Card To Compare With"
                        id="arrow"
                        src="arrow2.gif"
            />
</div>,
          <div
            className="shadow-card"
/>,
        ],
        "className": "displayOne",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "clicked": "clicked",
            "handleCompareCards": [Function],
            "schoolData": Object {
              "data": Object {
                "2004": 0.302,
                "2005": 0.267,
                "2006": 0.354,
                "2007": 0.392,
                "2008": 0.385,
                "2009": 0.39,
                "2010": 0.436,
                "2011": 0.489,
                "2012": 0.479,
                "2013": 0.488,
                "2014": 0.49,
              },
              "dataType": "Percent",
              "location": "ACADEMY 20",
            },
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="instructions"
>
                click another to compare
</p>,
              <img
                alt="Choose a Card To Compare With"
                id="arrow"
                src="arrow2.gif"
/>,
            ],
            "className": "instructions-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "click another to compare",
                "className": "instructions",
              },
              "ref": null,
              "rendered": "click another to compare",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Choose a Card To Compare With",
                "id": "arrow",
                "src": "arrow2.gif",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "shadow-card",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
